### PR TITLE
refactor: change mongo client mock 

### PR DESCRIPTION
**Pull request by eoaksnes**
## Why is this pull request needed?

* Makes it possible to run integration tests without spinning up MongoDB instance, instead use mongomock

## What does this pull request change?

* Add mongomock as dev dependency
* Add infrastructure fixtures
   * test_client - For direct access to mongo client
 * Inject repositories to endpoints 
   * Setup dependency_overrides for test_app for repositories during tests
* Add todo repository unit tests
* Add mongo database client unit tests

## Issues related to this change:

### CODE REVIEW

File: api/pyproject.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ isort = "^5.10.1"
 interrogate = "^1.5.0"
 bandit = "^1.7.4"
 pytest = "^7.1.2"
+mongomock = "4.1.2"
```
Comment (soofstad) [inline api/pyproject.toml:29]: Why is this locked?

File: api/src/features/todo/use_cases/delete_todo_by_id.py [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 
 
 class DeleteTodoByIdResponse(BaseModel):
     success: bool = Field(...)
 
 
-def delete_todo_use_case(id: str, todo_item_repository=TodoRepository()) -> DeleteTodoByIdResponse:
-    if not todo_item_repository.get(id):
+def delete_todo_use_case(id: str, todo_repository: TodoRepositoryInterface) -> DeleteTodoByIdResponse:
+    if not todo_repository.get(id):
```
Comment (soofstad) [inline api/src/features/todo/use_cases/delete_todo_by_id.py:11]: Not part of this PR. But IMO this should either return 404 or success=True

File: api/src/infrastructure/clients/mongodb/MongoDatabaseClient.py [hunk 3]
```diff
@@ -22,17 +23,22 @@
...
 
 class MongoDatabaseClient(ClientInterface[Dict, str]):
-    def __init__(self, collection_name: str, client: MongoClient = DEFAULT_MONGO_CLIENT):
-        self.handler = client[config.MONGODB_DATABASE]
-        self.collection: Collection = self.handler[collection_name]
+    def __init__(self, collection_name: str, database_name: str, client: MongoClient = DEFAULT_MONGO_CLIENT):
+        database: Database = client[database_name]
+        self.database = database
+        self.collection_name = collection_name
+        self.collection = database[collection_name]
```
Comment (soofstad) [inline api/src/infrastructure/clients/mongodb/MongoDatabaseClient.py:30]: Can these values be extracted from the client in an easy way?
Seems unnecessary to pass them when we are also passing the client

File: api/src/features/todo/use_cases/delete_todo_by_id.py (hunk 2 above)
Comment (eoaksnes) [reply api/src/features/todo/use_cases/delete_todo_by_id.py:11]: The mongo database client will throw a NotFoundException, so this should already be handled

File: api/pyproject.toml (hunk 1 above)
Comment (eoaksnes) [reply api/pyproject.toml]: Fixed to not locked

File: api/src/infrastructure/clients/mongodb/MongoDatabaseClient.py (hunk 3 above)
Comment (eoaksnes) [reply api/src/infrastructure/clients/mongodb/MongoDatabaseClient.py:17]: Nope, these can not be extracted from the client.

File: api/src/features/todo/use_cases/delete_todo_by_id.py (hunk 2 above)
Comment (soofstad) [reply api/src/features/todo/use_cases/delete_todo_by_id.py:11]: If that is the case, the entire if statement is redundant

**Review by soofstad — APPROVED** [review]

File: api/src/features/todo/use_cases/delete_todo_by_id.py (hunk 2 above)
Comment (eoaksnes) [reply api/src/features/todo/use_cases/delete_todo_by_id.py:11]: Should be fixed, adding a issue for this
